interfaces/repo.go

```diff
@@ -752,11 +767,15 @@ func isAutoConnectCandidate(plug *Plug, slot *Slot) bool {
 		// any content later
 		return false
 	}
-
 	// OS snap auto connect candidates
 	if slot.Snap.Type == snap.TypeOS && slot.Interface == plug.Interface {
 		return true
 	}
 
+	// FIXME: remove once we have assertions that provide this feature
+	if isSpecialAutoConnectSnap(plug.Snap) && slot.Interface == plug.Interface {
```

Comment:
Indeed! We should check the interface type and bail out if it is not compatible. Only after that we can use special-casing.